src/MongoDB.Analyzer/Core/Builders/BuildersVariablesResolver.cs

```diff
@@ -0,0 +1,396 @@
+                    variableDeclarator.ArgumentList,
+                    SyntaxFactory.EqualsValueClause(SyntaxFactory.Token(SyntaxKind.EqualsToken),
+                    rightHandAssignment.Left));
+
+                displayDiagnostics = false;
+            }
+
+            ProcessDeclarationOrAssignment(variableDeclarator, resolutionContext, dirtyVariables, displayDiagnostics);
+        }
+        else if (syntaxNode is AssignmentExpressionSyntax assignmentExpression)
+        {
+            var displayDiagnostics = true;
+            if (assignmentExpression.Right is AssignmentExpressionSyntax rightHandAssignment)
+            {
+                if (!TryProcessBuildersExpression(rightHandAssignment, resolutionContext, dirtyVariables))
+                {
+                    resolutionContext.CurrentVariablesValues.RemoveRange(ScrapeVariables(assignmentExpression));
+                    return false;
+                }
+
+                assignmentExpression = SyntaxFactory.AssignmentExpression(assignmentExpression.Kind(), assignmentExpression.Left, rightHandAssignment.Left);
+                displayDiagnostics = false;
+            }
+
+            ProcessDeclarationOrAssignment(assignmentExpression, resolutionContext, dirtyVariables, displayDiagnostics);
+        }
+        
+        return true;
+
+        bool ContainsConditionalExpression(SyntaxNode syntaxNode)
+        {
+            syntaxNode = syntaxNode switch
+            {
+                VariableDeclaratorSyntax @variableDeclarator => @variableDeclarator.Initializer.Value,
+                AssignmentExpressionSyntax @assignmentExpression => @assignmentExpression.Right,
+                _ => syntaxNode
+            };
+
+            syntaxNode = syntaxNode.TrimParenthesis();
+            return syntaxNode is SwitchExpressionSyntax || syntaxNode is ConditionalExpressionSyntax;
+        }
+
+        bool IsValidBuilderExpression(SyntaxNode syntaxNode)
+        {
+            var isValidBuilderExpression = false;
+
+            if (syntaxNode is VariableDeclaratorSyntax declarator)
+            {
+                isValidBuilderExpression = (semanticModel.GetSymbolInfo((declarator.Parent as VariableDeclarationSyntax).Type).Symbol as ITypeSymbol).IsBuilderDefinition() && !ContainsConditionalExpression(syntaxNode);
+            }
+            else if (syntaxNode is AssignmentExpressionSyntax assignment)
+            {
+                var leftHandSide = assignment.Left.TrimParenthesis();
+                isValidBuilderExpression = (leftHandSide is TupleExpressionSyntax tuple ? tuple.Arguments.Any(u => semanticModel.GetTypeInfo(u.TrimParenthesis()).Type.IsBuilderDefinition()) : semanticModel.GetTypeInfo(leftHandSide).Type.IsBuilderDefinition()) && !ContainsConditionalExpression(syntaxNode) && (assignment.Right is not AssignmentExpressionSyntax ? leftHandSide is IdentifierNameSyntax : true);
```

Comment:
`assignment.Right is AssignmentExpressionSyntax || leftHandSide is IdentifierNameSyntax` instead of 
`assignment.Right is not AssignmentExpressionSyntax ? leftHandSide is IdentifierNameSyntax : true`

Break long line to multi line:
```
isValidBuilderExpression = leftHandSide is TupleExpressionSyntax tuple ?
    tuple.Arguments.Any(u => semanticModel.GetTypeInfo(u.TrimParenthesis()).Type.IsBuilderDefinition()) :
    semanticModel.GetTypeInfo(leftHandSide).Type.IsBuilderDefinition() && !ContainsConditionalExpression(syntaxNode) && (assignment.Right is AssignmentExpressionSyntax || leftHandSide is IdentifierNameSyntax);
```

Reply:
Done